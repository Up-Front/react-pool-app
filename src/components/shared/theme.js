// this is the theme for styled components colors
export const theme = {
  fontColor: '#FFF',
  fontSize: '16px',
  background: '#243949',
  borderColor: '#FFF',
  backgroundImage: 'linear-gradient(to top, #243949 70%, #517fa4 100%)',

  headerBackgroundColor: 'rgba(0, 0, 0, 0.7)',
  headerFontColor: '#FFF',

  buttonColor: 'DarkOrange',
  buttonFontColor: '#FFF',

  //match
  matchBackgroundColor: '#243949',
  contestedMatchBackgroundColor: 'red',

  //delete swipe
  deleteBackgroundColor: 'red',

  //user / competitor
  statusOnlineColor: 'green',
  statusOfflineColor: 'orange',
  statusUnknownColor: 'lightgrey',
};
