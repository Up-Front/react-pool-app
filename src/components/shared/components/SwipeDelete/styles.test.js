import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { theme } from './../../theme';
import {
  DeleteLayer,
  DeleteLayerLeft,
  DeleteLayerRight,
  SwipeContentWrapper,
  SwipeContent,
} from './styles';

describe('DeleteLayer component', () => {
  test('renders normally', () => {
    const wrapper = shallow(<DeleteLayer theme={theme} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule('background-color', 'red');
  });
});

describe('DeleteLayerLeft component', () => {
  test('renders normally', () => {
    const wrapper = shallow(<DeleteLayerLeft />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('DeleteLayerRight component', () => {
  test('renders normally', () => {
    const wrapper = shallow(<DeleteLayerRight />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('SwipeContentWrapper component', () => {
  test('renders normally', () => {
    const wrapper = shallow(<SwipeContentWrapper />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('SwipeContent component', () => {
  test('renders normally', () => {
    const wrapper = shallow(<SwipeContent />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders normally swiped left', () => {
    const wrapper = shallow(<SwipeContent deleteDirection="left" />);
    expect(wrapper).toHaveStyleRule('left', '-100%!important');
  });

  test('renders normally swiped right', () => {
    const wrapper = shallow(<SwipeContent deleteDirection="right" />);
    expect(wrapper).toHaveStyleRule('left', '100%!important');
  });

  test('renders normally swipe cancelled', () => {
    const wrapper = shallow(<SwipeContent deleteCancel={true} />);
    expect(wrapper).toHaveStyleRule('left', '0!important');
  });
});
