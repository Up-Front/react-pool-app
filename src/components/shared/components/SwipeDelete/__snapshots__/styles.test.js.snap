// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteLayer component renders normally 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div
    theme={
        Object {
            "background": "#243949",
            "backgroundImage": "linear-gradient(to top, #243949 70%, #517fa4 100%)",
            "borderColor": "#FFF",
            "buttonColor": "DarkOrange",
            "buttonFontColor": "#FFF",
            "contestedMatchBackgroundColor": "red",
            "deleteBackgroundColor": "red",
            "fontColor": "#FFF",
            "fontSize": "16px",
            "headerBackgroundColor": "rgba(0, 0, 0, 0.7)",
            "headerFontColor": "#FFF",
            "matchBackgroundColor": "#243949",
            "movementDown": "#da1418",
            "movementUp": "#66b75b",
            "statusOfflineColor": "orange",
            "statusOnlineColor": "green",
            "statusUnknownColor": "lightgrey",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bwzfXH bgfRJu",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bwzfXH bgfRJu",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`DeleteLayerLeft component renders normally 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-htpNat AHDt",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-htpNat AHDt",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`DeleteLayerRight component renders normally 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bxivhb gAgXim",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bxivhb gAgXim",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SwipeContent component renders normally 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-EHOje ceSwgL",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-EHOje ceSwgL",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SwipeContentWrapper component renders normally 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-ifAKCX dItWzQ",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-ifAKCX dItWzQ",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
